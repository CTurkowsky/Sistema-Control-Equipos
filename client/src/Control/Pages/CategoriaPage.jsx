import { Button, Grid, TextField, Alert } from '@mui/material';
import { FormLayout } from '../../Control/Layout/FormLayout';
import { useFormik } from 'formik';
import * as YUP from 'yup';
import { createCategoriaRequest } from '../../api/categoria.api';
import { useCategorias } from '../../hooks';
import Swal from 'sweetalert2';
export const CategoriaPage = () => {
  const { categorias } = useCategorias();
  const formik = useFormik({
    initialValues: {
      categoria: '',
    },
    validationSchema: YUP.object({
      categoria: YUP.string()
        .min(3, 'El nombre tener mas de 3 caracteres')
        .max(20, 'El nombre debe tener maximo  20 caracteres')
        .required('El nombre es requerido'),
    }),

    onSubmit: async (values) => {
      console.log(values);
      try {
        const soloCategorias = categorias.map(
          (categoria) => categoria.categoria
        );
        const matches = soloCategorias.filter((element) => {
          return values.categoria.toLowerCase() == element.toLowerCase();
        });
        if (matches.length > 0)
          return Swal.fire({
            title: 'Error!',
            text: 'Categoria ya existe',
            icon: 'error',
            confirmButtonText: 'Aceptar',
          });
        const response = await createCategoriaRequest(values);
        Swal.fire({
          title: 'Exitoso!',
          text: 'Se ha registrado una categoria',
          icon: 'success',
          confirmButtonText: 'Aceptar',
        });
        formik.resetForm();
      } catch (error) {
        console.log(error);
      }
    },
  });

  return (
    <>
      <FormLayout title='Registro Categoria'>
        <form onSubmit={formik.handleSubmit}>
          <Grid container>
            <Grid item xs={12} sx={{ mt: 2 }}>
              <TextField
                label='Nombre'
                type='text'
                placeholder='Nombre Categoria'
                fullWidth
                name='categoria'
                value={formik.values.categoria}
                onChange={formik.handleChange}
                onBlur={formik.handleBlur}
              />
              {formik.touched.categoria && formik.errors.categoria ? (
                <Alert sx={{ mt: 2 }} severity='error'>
                  {formik.errors.categoria}
                </Alert>
              ) : null}
            </Grid>
            <Grid container spacing={2} sx={{ mb: 2, mt: 1 }}>
              <Grid item xs={12}>
                <Button
                  variant='contained'
                  fullWidth
                  type='submit'
                  sx={{ color: '#ffff' }}
                >
                  Registrar
                </Button>
              </Grid>
            </Grid>
          </Grid>
        </form>
      </FormLayout>
    </>
  );
};
