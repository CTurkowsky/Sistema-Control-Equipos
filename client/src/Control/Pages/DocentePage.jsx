import {
  Button,
  Grid,
  TextField,
  Alert,
  Select,
  MenuItem,
  InputLabel,
} from '@mui/material';
import { FormLayout } from '../Layout/FormLayout';
import { useFormik } from 'formik';
import * as YUP from 'yup';
import { createDocenteRequest } from '../../api/docente.api';
import { useDocentes } from '../../hooks';
import Swal from 'sweetalert2';
export const DocentePage = () => {
  const { docentes } = useDocentes();
  const formik = useFormik({
    initialValues: {
      nombre: '',
      apellidoPaterno: '',
      apellidoMaterno: '',
      numeroCelular: '',
      grado: '',
      seccion: '',
    },
    validationSchema: YUP.object({
      nombre: YUP.string()
        .min(3, 'El nombre tener mas de 3 caracteres')
        .max(20, 'El nombre debe tener maximo  20 caracteres')
        .required('El nombre es requerido'),
      apellidoPaterno: YUP.string()
        .min(3, 'El apellido paterno debe tener mas de 3 caracteres')
        .max(50, 'El apellido paterno debe tener maximo  50 caracteres')
        .required('La descripcion es requerido'),
      apellidoMaterno: YUP.string()
        .min(3, 'El apellido materno debe tener mas de 3 caracteres')
        .max(50, 'El apellido materno debe tener maximo  50 caracteres')
        .required('El numero es requerido'),
      numeroCelular: YUP.number()
        .positive('El numero debe ser valido')
        .min(9, 'Ingrese 9 digitos')
        .required('El numero de celular es requerido'),

      grado: YUP.string().required('El grado es requerido'),
      seccion: YUP.string().required('La seccion es requerida'),
    }),

    onSubmit: async (values) => {
      console.log(values);
      try {
        if (
          docentes.find(
            (docente) =>
              docente.apellidoPaterno === values.apellidoPaterno &&
              docente.apellidoMaterno === values.apellidoMaterno &&
              docente.nombre === values.nombre
          )
        )
          return Swal.fire({
            title: 'Error!',
            text: 'Docente ya existe',
            icon: 'error',
            confirmButtonText: 'Aceptar',
          });
        const response = await createDocenteRequest(values);
        formik.resetForm();
        return Swal.fire({
          title: 'Success!',
          text: 'Se ha registrado un docente',
          icon: 'success',
          confirmButtonText: 'Aceptar',
        });
      } catch (error) {
        console.log(error);
      }
    },
  });

  return (
    <>
      <FormLayout title='Docente'>
        <form onSubmit={formik.handleSubmit}>
          <Grid container>
            <Grid item xs={12} sx={{ mt: 2 }}>
              <TextField
                label='Nombre'
                type='text'
                placeholder='Nombre del Docente'
                fullWidth
                name='nombre'
                value={formik.values.nombre}
                onChange={formik.handleChange}
                onBlur={formik.handleBlur}
              />
              {formik.touched.nombre && formik.errors.nombre ? (
                <Alert sx={{ mt: 2 }} severity='error'>
                  {formik.errors.nombre}
                </Alert>
              ) : null}
            </Grid>
            <Grid item xs={12} sx={{ mt: 2 }}>
              <TextField
                label='Apellido Paterno'
                type='text'
                placeholder='Apellido Paterno'
                fullWidth
                name='apellidoPaterno'
                value={formik.values.apellidoPaterno}
                onChange={formik.handleChange}
                onBlur={formik.handleBlur}
              />
              {formik.touched.apellidoPaterno &&
              formik.errors.apellidoPaterno ? (
                <Alert sx={{ mt: 2 }} severity='error'>
                  {formik.errors.apellidoPaterno}
                </Alert>
              ) : null}
            </Grid>
            <Grid item xs={12} sx={{ mt: 2 }}>
              <TextField
                label='Apellido Materno'
                type='text'
                placeholder='Apellido Materno'
                fullWidth
                name='apellidoMaterno'
                value={formik.values.apellidoMaterno}
                onChange={formik.handleChange}
                onBlur={formik.handleBlur}
              />
              {formik.touched.apellidoMaterno &&
              formik.errors.apellidoMaterno ? (
                <Alert sx={{ mt: 2 }} severity='error'>
                  {formik.errors.apellidoMaterno}
                </Alert>
              ) : null}
            </Grid>
            <Grid item xs={12} sx={{ mt: 2 }}>
              <TextField
                label='Numero Celular'
                type='text'
                placeholder='999999999'
                fullWidth
                name='numeroCelular'
                value={formik.values.numeroCelular}
                onChange={formik.handleChange}
                onBlur={formik.handleBlur}
              />
              {formik.touched.numeroCelular && formik.errors.numeroCelular ? (
                <Alert sx={{ mt: 2 }} severity='error'>
                  {formik.errors.numeroCelular}
                </Alert>
              ) : null}
            </Grid>
            <Grid item xs={12} sx={{ mt: 2 }}>
              <InputLabel>Grado</InputLabel>
              <Select
                onChange={formik.handleChange}
                value={formik.values.grado}
                onBlur={formik.handleBlur}
                label='Grado'
                name='grado'
                fullWidth
              >
                <MenuItem value={1}>1º</MenuItem>
                <MenuItem value={2}>2º</MenuItem>
                <MenuItem value={3}>3º</MenuItem>
                <MenuItem value={4}>4º</MenuItem>
                <MenuItem value={5}>5º</MenuItem>
                <MenuItem value={6}>6º</MenuItem>
              </Select>
              {formik.touched.grado && formik.errors.grado ? (
                <Alert sx={{ mt: 2 }} severity='error'>
                  {formik.errors.grado}
                </Alert>
              ) : null}
            </Grid>

            <Grid item xs={12} sx={{ mt: 2 }}>
              <InputLabel>Seccion</InputLabel>
              <Select
                onChange={formik.handleChange}
                value={formik.values.seccion}
                onBlur={formik.handleBlur}
                label='Seccion'
                name='seccion'
                fullWidth
              >
                <MenuItem value={'Primaria'}>Primaria</MenuItem>
                <MenuItem value={'Secundaria'}>Secundaria</MenuItem>
              </Select>
              {formik.touched.seccion && formik.errors.seccion ? (
                <Alert sx={{ mt: 2 }} severity='error'>
                  {formik.errors.seccion}
                </Alert>
              ) : null}
            </Grid>
            <Grid container spacing={2} sx={{ mb: 2, mt: 1 }}>
              <Grid item xs={12}>
                <Button
                  variant='contained'
                  fullWidth
                  type='submit'
                  sx={{ color: '#ffff' }}
                >
                  Registrar
                </Button>
              </Grid>
            </Grid>
          </Grid>
        </form>
      </FormLayout>
    </>
  );
};
