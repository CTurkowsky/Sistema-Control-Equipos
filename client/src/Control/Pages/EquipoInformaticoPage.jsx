import {
  Button,
  Grid,
  TextField,
  Alert,
  Select,
  MenuItem,
  InputLabel,
} from '@mui/material';
import { FormLayout } from '../Layout/FormLayout';
import { createEquipoRequest } from '../../api/equipoInformatico.api';
import { useMarcas, useCategorias, useEquipos } from '../../hooks';
import { useFormik } from 'formik';
import * as YUP from 'yup';
import Swal from 'sweetalert2';
export const EquipoInformaticoPage = () => {
  const { marcas } = useMarcas();
  const { categorias } = useCategorias();
  const { equipos } = useEquipos();
  const formik = useFormik({
    initialValues: {
      nombre: '',
      descripcion: '',
      categoria: '',
      numeroSerie: '',
      estado: '',
      marca: '',
      disponibilidad: 'Si',
    },
    validationSchema: YUP.object({
      nombre: YUP.string()
        .min(3, 'El nombre tener mas de 3 caracteres')
        .max(20, 'El nombre debe tener maximo  20 caracteres')
        .required('El nombre es requerido'),
      descripcion: YUP.string()
        .min(3, 'La descripcion debe tener mas de 3 caracteres')
        .max(50, 'La descripcion debe tener maximo  50 caracteres')
        .required('La descripcion es requerido'),
      numeroSerie: YUP.string()
        .min(3, 'El numero de serie debe tener mas de 3 caracteres')
        .max(50, 'El numero de serie debe tener maximo  50 caracteres')
        .required('El numero es requerido'),
      marca: YUP.string().required('La marca es requerida'),
      estado: YUP.string().required('El estado  es requerida'),
      categoria: YUP.string().required('La categoria es requerida'),
    }),

    onSubmit: async (values) => {
      console.log(values);
      try {
        if (equipos.find((equipo) => equipo.numeroSerie === values.numeroSerie))
          return Swal.fire({
            title: 'Error!',
            text: 'Codigo ya existe',
            icon: 'error',
            confirmButtonText: 'Aceptar',
          });
        const response = await createEquipoRequest(values);
        Swal.fire({
          title: 'Exitoso!',
          text: 'Se ha registrado un producto',
          icon: 'success',
          confirmButtonText: 'Aceptar',
        });
        formik.resetForm();
      } catch (error) {
        console.log(error);
      }
    },
  });

  return (
    <>
      <FormLayout title='Equipo Informatico'>
        <form onSubmit={formik.handleSubmit}>
          <Grid container>
            <Grid item xs={12} sx={{ mt: 2 }}>
              <TextField
                label='Nombre'
                type='text'
                placeholder='Nombre del equipo'
                fullWidth
                name='nombre'
                value={formik.values.nombre}
                onChange={formik.handleChange}
                onBlur={formik.handleBlur}
              />
              {formik.touched.nombre && formik.errors.nombre ? (
                <Alert sx={{ mt: 2 }} severity='error'>
                  {formik.errors.nombre}
                </Alert>
              ) : null}
            </Grid>
            <Grid item xs={12} sx={{ mt: 2 }}>
              <TextField
                label='Descripcion'
                type='text'
                placeholder='Descripcion'
                fullWidth
                name='descripcion'
                value={formik.values.descripcion}
                onChange={formik.handleChange}
                onBlur={formik.handleBlur}
              />
              {formik.touched.descripcion && formik.errors.descripcion ? (
                <Alert sx={{ mt: 2 }} severity='error'>
                  {formik.errors.descripcion}
                </Alert>
              ) : null}
            </Grid>
            <Grid item xs={12} sx={{ mt: 2 }}>
              <InputLabel>Categoria</InputLabel>
              <Select
                fullWidth
                label='Categoria'
                name='categoria'
                value={formik.values.categoria}
                onChange={formik.handleChange}
                onBlur={formik.handleBlur}
              >
                {categorias.map((categoria) => (
                  <MenuItem
                    key={categoria.idCategoria}
                    value={categoria.idCategoria}
                  >
                    {categoria.categoria}
                  </MenuItem>
                ))}
              </Select>
              {formik.touched.categoria && formik.errors.categoria ? (
                <Alert sx={{ mt: 2 }} severity='error'>
                  {formik.errors.categoria}
                </Alert>
              ) : null}
            </Grid>
            <Grid item xs={12} sx={{ mt: 2 }}>
              <TextField
                label='Numero Serie'
                type='text'
                placeholder='Numero Serie'
                fullWidth
                name='numeroSerie'
                value={formik.values.numeroSerie}
                onChange={formik.handleChange}
                onBlur={formik.handleBlur}
              />

              {formik.touched.numeroSerie && formik.errors.numeroSerie ? (
                <Alert sx={{ mt: 2 }} severity='error'>
                  {formik.errors.numeroSerie}
                </Alert>
              ) : null}
            </Grid>
            <Grid item xs={12} sx={{ mt: 2 }}>
              <InputLabel>Estado</InputLabel>
              <Select
                onChange={formik.handleChange}
                value={formik.values.estado}
                onBlur={formik.handleBlur}
                label='Estado'
                name='estado'
                fullWidth
              >
                <MenuItem value={'Operativo'}>Operativo</MenuItem>
                <MenuItem value={'Necesita Reparacion'}>
                  Necesita Reparacion
                </MenuItem>
              </Select>
              {formik.touched.estado && formik.errors.estado ? (
                <Alert sx={{ mt: 2 }} severity='error'>
                  {formik.errors.estado}
                </Alert>
              ) : null}
            </Grid>
            <Grid item xs={12} sx={{ mt: 2 }}>
              <InputLabel>Marca</InputLabel>
              <Select
                fullWidth
                label='Marca'
                name='marca'
                value={formik.values.marca}
                onChange={formik.handleChange}
                onBlur={formik.handleBlur}
              >
                {marcas.map((marca) => (
                  <MenuItem key={marca.idMarca} value={marca.idMarca}>
                    {marca.marca}
                  </MenuItem>
                ))}
              </Select>
              {formik.touched.marca && formik.errors.marca ? (
                <Alert sx={{ mt: 2 }} severity='error'>
                  {formik.errors.marca}
                </Alert>
              ) : null}
            </Grid>
            <Grid container spacing={2} sx={{ mb: 2, mt: 1 }}>
              <Grid item xs={12}>
                <Button
                  variant='contained'
                  fullWidth
                  type='submit'
                  sx={{ color: '#ffff' }}
                >
                  Registrar
                </Button>
              </Grid>
            </Grid>
          </Grid>
        </form>
      </FormLayout>
    </>
  );
};
