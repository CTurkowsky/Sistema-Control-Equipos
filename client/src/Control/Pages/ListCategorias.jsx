import { useCategorias } from '../../hooks';
import { Typography, Box, CardContent, Button, Grid } from '@mui/material';
import { ListLayout } from '../Layout/ListLayout';
export const ListCategorias = () => {
  const { categorias, deleteCategoria } = useCategorias();


  
  return (
    <Grid
       sx={{
          display: 'flex',
          flexFlow: 'row wrap',
          justifyContent: 'center',
          alignContent: 'center',
          mx:  'auto',
        backgroundColor: 'primary.main',
        }} >

      {categorias.map((categoria) => (
        <Box key={categoria.idCategoria} sx={{ margin: 4 }}>

          <CardContent>
            <Grid
              alignItems='center'
              sx={{
                minHeight: '5vh',
                backgroundColor: '#eee',
                padding: 4,
                width: 500,
                height: 200,
                borderRadius: 7,
              justifyContent:'center',
                alignContent: 'center'
              }}
            >
              <Typography variant='h4' component='div'>
                <a>Categoria: </a>
                {categoria.categoria}
              </Typography>
              <Button variant='contained' onClick={() => deleteCategoria(categoria.idCategoria)}>Eliminar</Button>
            </Grid>
          </CardContent>
        </Box>
      ))}

    </Grid>
  );
};

