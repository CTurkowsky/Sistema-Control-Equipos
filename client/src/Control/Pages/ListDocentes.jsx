import { useDocentes } from '../../hooks';
import { Typography, Box, CardContent, Button, Grid } from '@mui/material';
import { ListLayout } from '../Layout/ListLayout';
export const ListDocentes = () => {
  const { docentes } = useDocentes();
  return (
    <Grid
       sx={{
          display: 'flex',
          flexFlow: 'row wrap',
          justifyContent: 'center',
          alignContent: 'center',
          mx:  'auto',
        backgroundColor: 'primary.main',
        }} >

      {docentes.map((docente) => (
        <Box key={docente.idDocente} sx={{ margin: 4 }}>

          <CardContent>
            <Grid
              alignItems='center'
              sx={{
                minHeight: '5vh',
                backgroundColor: '#eee',
                padding: 4,
                width: 500,
                height: 200,
                borderRadius: 7,
              justifyContent:'center',
                alignContent: 'center'
              }}
            >
              <Typography variant='h4' component='div'>
                <a>Docente: </a>
                {docente.nombre}{' '}{docente.apellidoPaterno}{' '}{docente.apellidoMaterno}
              </Typography>
            </Grid>
          </CardContent>
        </Box>
      ))}

    </Grid>
  );
};



