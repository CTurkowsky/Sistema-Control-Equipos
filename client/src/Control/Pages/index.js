export * from './CategoriaPage'
export * from './MarcaPage';
export * from './EquipoInformaticoPage';
export * from './IncidenciaPage';
export * from './ListEquipos';
export * from './ListDocentes';
export * from './EquipoPrestamo';
export * from './ListEquipoPrestamo';
export * from './ListPrestamos';
export * from './ListIncidencia';
export * from './ListCategorias';
export * from './ListMarcas';
export * from './PrestamoPage';
export * from './DocentePage';