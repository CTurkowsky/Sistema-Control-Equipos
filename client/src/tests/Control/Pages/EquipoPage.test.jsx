import { render, screen } from '@testing-library/react'
import { EquipoInformaticoPage } from '../../../Control/Pages/EquipoInformaticoPage'

describe('Pruebas en <EquipoInformaticoPage/>', () => { 

    test('Debe de hacer match con el snaptshot', ()=>{
        const { container } = render(<EquipoInformaticoPage/>)
       expect(container).toMatchSnapshot();
    })

    test('Debe concidir con los nombres',()=> {
        const { container } = render(<EquipoInformaticoPage/>)
        expect(screen.getByRole("textbox", {name: /nombre/i}))
        expect(screen.getByRole("textbox", {name: /descripcion/i}))
        // expect(screen.getByRole("textbox", {name: /numeroSerie/i}))
        // expect(screen.getByRole("textbox", {name: /estado/i}))
        // expect(screen.getByRole('combobox', {name: /marca/i}))
        // expect(screen.getByRole("textbox", {name: /categoria/i}))
    })
})