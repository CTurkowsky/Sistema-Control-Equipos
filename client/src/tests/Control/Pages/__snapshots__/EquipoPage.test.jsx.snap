// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <EquipoInformaticoPage/> Debe de hacer match con el snaptshot 1`] = `
<div>
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-column css-ro1yt6-MuiGrid-root"
  >
    <div
      class="MuiGrid-root css-1tepu03-MuiGrid-root"
    >
      <h5
        class="MuiTypography-root MuiTypography-h5 css-1jcm1rz-MuiTypography-root"
      >
        Equipo Informatico
         
      </h5>
      <form>
        <div
          class="MuiGrid-root MuiGrid-container css-11lq3yg-MuiGrid-root"
        >
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1hzi63y-MuiGrid-root"
          >
            <div
              class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
            >
              <label
                class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
                data-shrink="false"
                for=":r0:"
                id=":r0:-label"
              >
                Nombre
              </label>
              <div
                class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
              >
                <input
                  aria-invalid="false"
                  class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                  id=":r0:"
                  name="nombre"
                  placeholder="Nombre del equipo"
                  type="text"
                  value=""
                />
                <fieldset
                  aria-hidden="true"
                  class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    class="css-yjsfm1"
                  >
                    <span>
                      Nombre
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1hzi63y-MuiGrid-root"
          >
            <div
              class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
            >
              <label
                class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
                data-shrink="false"
                for=":r1:"
                id=":r1:-label"
              >
                Descripcion
              </label>
              <div
                class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
              >
                <input
                  aria-invalid="false"
                  class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                  id=":r1:"
                  name="descripcion"
                  placeholder="Descripcion"
                  type="text"
                  value=""
                />
                <fieldset
                  aria-hidden="true"
                  class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    class="css-yjsfm1"
                  >
                    <span>
                      Descripcion
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1hzi63y-MuiGrid-root"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-animated MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-animated css-9npbnl-MuiFormLabel-root-MuiInputLabel-root"
            >
              Categoria
            </label>
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth  css-1ualgfl-MuiInputBase-root-MuiOutlinedInput-root-MuiSelect-root"
            >
              <div
                aria-expanded="false"
                aria-haspopup="listbox"
                aria-labelledby="mui-component-select-categoria"
                class="MuiSelect-select MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input css-11u53oe-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
                id="mui-component-select-categoria"
                role="button"
                tabindex="0"
              >
                <span
                  class="notranslate"
                >
                  ​
                </span>
              </div>
              <input
                aria-hidden="true"
                class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
                name="categoria"
                tabindex="-1"
                value=""
              />
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
                data-testid="ArrowDropDownIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M7 10l5 5 5-5z"
                />
              </svg>
              <fieldset
                aria-hidden="true"
                class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="css-yjsfm1"
                >
                  <span>
                    Categoria
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1hzi63y-MuiGrid-root"
          >
            <div
              class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
            >
              <label
                class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
                data-shrink="false"
                for=":r2:"
                id=":r2:-label"
              >
                Numero Serie
              </label>
              <div
                class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
              >
                <input
                  aria-invalid="false"
                  class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                  id=":r2:"
                  name="numeroSerie"
                  placeholder="Numero Serie"
                  type="text"
                  value=""
                />
                <fieldset
                  aria-hidden="true"
                  class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    class="css-yjsfm1"
                  >
                    <span>
                      Numero Serie
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1hzi63y-MuiGrid-root"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-animated MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-animated css-9npbnl-MuiFormLabel-root-MuiInputLabel-root"
            >
              Estado
            </label>
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth  css-1ualgfl-MuiInputBase-root-MuiOutlinedInput-root-MuiSelect-root"
            >
              <div
                aria-expanded="false"
                aria-haspopup="listbox"
                aria-labelledby="mui-component-select-estado"
                class="MuiSelect-select MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input css-11u53oe-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
                id="mui-component-select-estado"
                role="button"
                tabindex="0"
              >
                <span
                  class="notranslate"
                >
                  ​
                </span>
              </div>
              <input
                aria-hidden="true"
                class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
                name="estado"
                tabindex="-1"
                value=""
              />
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
                data-testid="ArrowDropDownIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M7 10l5 5 5-5z"
                />
              </svg>
              <fieldset
                aria-hidden="true"
                class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="css-yjsfm1"
                >
                  <span>
                    Estado
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1hzi63y-MuiGrid-root"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-animated MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-animated css-9npbnl-MuiFormLabel-root-MuiInputLabel-root"
            >
              Marca
            </label>
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth  css-1ualgfl-MuiInputBase-root-MuiOutlinedInput-root-MuiSelect-root"
            >
              <div
                aria-expanded="false"
                aria-haspopup="listbox"
                aria-labelledby="mui-component-select-marca"
                class="MuiSelect-select MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input css-11u53oe-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
                id="mui-component-select-marca"
                role="button"
                tabindex="0"
              >
                <span
                  class="notranslate"
                >
                  ​
                </span>
              </div>
              <input
                aria-hidden="true"
                class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
                name="marca"
                tabindex="-1"
                value=""
              />
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
                data-testid="ArrowDropDownIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M7 10l5 5 5-5z"
                />
              </svg>
              <fieldset
                aria-hidden="true"
                class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="css-yjsfm1"
                >
                  <span>
                    Marca
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 css-h80lhz-MuiGrid-root"
          >
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
            >
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-fullWidth MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-fullWidth css-dxhjt-MuiButtonBase-root-MuiButton-root"
                tabindex="0"
                type="submit"
              >
                Registrar
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
            </div>
          </div>
        </div>
      </form>
    </div>
  </div>
</div>
`;
